File: Beta Progrmming/1121-Threeparen.cpp
#include <bits/stdc++.h>
using namespace std;
stack<char> st;
char a[100100];
int main(){
    int len,q,i;
    scanf("%d",&q);
    while (q--)
    {
        scanf(" %s",a);
        len = strlen(a);
        for(i=0;i<len;i++){
            if(a[i]=='(' || a[i]=='[' || a[i]=='{') st.push(a[i]);
            else if(a[i]==')' && st.top()=='(') st.pop();
            else if(a[i]==']' && st.top()=='[') st.pop();
            else if(a[i]=='}' && st.top()=='{') st.pop();
            else st.push(a[i]);
        }
        if(st.empty()) printf("yes\n");
        else printf("no\n");
        while (!st.empty())
        {
            st.pop();
        }
        
    }
    
    return 0;
}

File: Python/1121_Threeparen.py
q = int(input())
o = ['(','[','{']
c = [')',']','}']
while q :
    q -= 1
    a = input()
    st = []
    for x in a :
        if x in o : st.append(x)
        elif x in c :
            if len(st) > 0 :
                if x == ')' and st[len(st)-1] == '(' : st.pop()
                elif x == ']' and st[len(st)-1] == '[' : st.pop()
                elif x == '}' and st[len(st)-1] == '{' : st.pop()
                else : 
                    st.append(x)
                    break
            else :
                st.append(x)
                break
    if len(st) == 0 : print('yes')
    else : print('no')
    st.clear()